xrpl/wallet.go

```diff
@@ -65,55 +70,55 @@ func NewWalletFromSecret(seed string) (Wallet, error) {
 
 // // Derives a wallet from a bip39 or RFC1751 mnemonic (Defaults to bip39).
 // // Returns a Wallet object. If an error occurs, it will be returned.
-// func NewWalletFromMnemonic(mnemonic string) (*Wallet, error) {
-// 	// Validate the mnemonic
-// 	if !bip39.IsMnemonicValid(mnemonic) {
-// 		return nil, errors.New("invalid mnemonic")
-// 	}
-
-// 	// Generate seed from mnemonic
-// 	seed := bip39.NewSeed(mnemonic, "")
-
-// 	// Derive the master key
-// 	masterKey, err := bip32.NewMasterKey(seed)
-// 	if err != nil {
-// 		return nil, err
-// 	}
-
-// 	// Derive the key using the path m/44'/144'/0'/0/0
-// 	path := []uint32{
-// 		44 + bip32.FirstHardenedChild,
-// 		144 + bip32.FirstHardenedChild,
-// 		bip32.FirstHardenedChild,
-// 		0,
-// 		0,
-// 	}
-
-// 	key := masterKey
-// 	for _, childNum := range path {
-// 		key, err = key.NewChildKey(childNum)
-// 		if err != nil {
-// 			return nil, err
-// 		}
-// 	}
-
-// 	// Convert the private key to the format expected by the XRPL library
-// 	privKey := strings.ToUpper(hex.EncodeToString(key.Key))
-// 	pubKey := strings.ToUpper(hex.EncodeToString(key.PublicKey().Key))
-
-// 	// Derive classic address
-// 	classicAddr, err := keypairs.DeriveClassicAddress(pubKey)
-// 	if err != nil {
-// 		return nil, err
-// 	}
-
-// 	return &Wallet{
-// 		PublicKey:      pubKey,
-// 		PrivateKey:     fmt.Sprintf("00%s", privKey),
-// 		ClassicAddress: classicAddr,
-// 		Seed:           "", // We don't have the seed in this case
-// 	}, nil
-// }
+func NewWalletFromMnemonic(mnemonic string) (*Wallet, error) {
+	// Validate the mnemonic
+	if !bip39.IsMnemonicValid(mnemonic) {
+		return nil, errors.New("invalid mnemonic")
+	}
+
+	// Generate seed from mnemonic
+	seed := bip39.NewSeed(mnemonic, "")
+
+	// Derive the master key
+	masterKey, err := bip32.NewMasterKey(seed)
+	if err != nil {
+		return nil, err
+	}
+
+	// Derive the key using the path m/44'/144'/0'/0/0
+	path := []uint32{
+		44 + bip32.FirstHardenedChild,
+		144 + bip32.FirstHardenedChild,
+		bip32.FirstHardenedChild,
+		0,
+		0,
+	}
+
+	key := masterKey
+	for _, childNum := range path {
+		key, err = key.NewChildKey(childNum)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Convert the private key to the format expected by the XRPL library
+	privKey := strings.ToUpper(hex.EncodeToString(key.Key))
+	pubKey := strings.ToUpper(hex.EncodeToString(key.PublicKey().Key))
+
+	// Derive classic address
+	classicAddr, err := keypairs.DeriveClassicAddress(pubKey)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Wallet{
+		PublicKey:      pubKey,
+		PrivateKey:     fmt.Sprintf("00%s", privKey),
```

Comment:
So this generation from a mnemonic only generates a secp256k1 key ? Is it possible to generate an ed25519 also ?

Reply:
By default it should be secp256k1 to match the rippled `wallet_propose` default algorithm. If needed we can add it later